handlebars/src/main/java/com/github/jknack/handlebars/internal/HelperResolver.java

```diff
@@ -279,4 +288,16 @@ protected void collectReferenceParameters(final Collection<String> result) {
     }
   }
 
+  @Override
+  protected void collectAllParameters(final Collection<Param> result) {
+    for (Param param : params) {
+      if (param instanceof VarParam) {
+        ((VarParam) param).fn.collectAllParameters(result);
```

Comment:
is this collecting nested hbars, for example `{{#if (anotherHelper) }}`? or is this doing something else?

Reply:
doesn't seem to currently; an input of `{{capitalize (lower firstName)}}` returns `TagWithParams(capitalize, firstName, TagType.VAR)`

If I include `TagType.SUB_EXPRESSION`, it does work. Let me change the implementation to `template.collectWithParameters(TagType.values())`